src/logic/useWalkTime.ts

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react'
+import { isEqual } from 'lodash'
+import service from '../service'
+import { Coordinates } from '@entur/sdk'
+import { useSettingsContext } from '../settings'
+import { StopPlaceWithDepartures } from '../types'
+import { usePrevious } from '../utils'
+
+async function getWalkTime(
+    stopPlaces: StopPlaceWithDepartures[],
+    from: Coordinates,
+): Promise<Array<{ stopId: string; walkTime: number }>> {
+    const travelTimes = Promise.all(
+        stopPlaces.map(
+            async (stopPlace) =>
+                await service
+                    .getTripPatterns({
+                        from: {
+                            name: 'pin',
+                            coordinates: from,
+                        },
+                        to: {
+                            name: stopPlace.name,
+                            place: stopPlace.id,
+                        },
+                        modes: ['foot'],
+                        limit: 1,
+                    })
+                    .then((result) => ({
+                        stopId: stopPlace.id,
+                        walkTime: result[0].duration,
+                    })),
```

Comment:
Det bør vere ei håndtering av feil her (`.catch(...)`). Kanskje returnere `null` dersom det kræsjer, og så filtere lista i etterkant for å fjerne null-verdiar?
```
    .catch((error) => null)
  ),
).filter(notNullOrUndefined)
```

Reply:
Gjort det nå, men ser ikke ut som typescript helt forstår at null blir filtrert bort. Måtte legge til null i interfacet og return typen... ```Array<{ stopId: string; walkTime: number } | null>```